### DISCUSSION TITLE
pdf.js eval security risk

### DISCUSSION THREAD

**Discussion by TahmidU** [wojtekmaj/react-pdf#1757] [Q&A]
I've been using this package with `vite` and noticed this warning when I run `npm i`, `npm run build`...

```
node_modules/pdfjs-dist/build/pdf.js (1982:23): Use of eval in "node_modules/pdfjs-dist/build/pdf.js" is strongly discouraged as it poses security risks and may cause issues with minification.
✓ 107 modules transformed.
computing gzip size (2)...[vite-plugin-static-copy] Copied 2 items.
dist/index.html                             0.42 kB │ gzip:   0.28 kB
dist/assets/pdf.worker.min-DKQKFyKK.js  1,087.21 kB
dist/assets/index-Ej-CiX3I.css              9.75 kB │ gzip:   2.19 kB
dist/assets/index-CdrYiMFw.js             528.54 kB │ gzip: 156.82 kB
```

You get this warning when you use `vite` with this package. Not sure if this also happens with `webpack` and the others. The error above was taken from the vite sample project (`sample/vite`)

I understand that the 'problem' is coming from `pdf.js` itself but I thought it'll be interesting to ask this community. Should I be concerned about the security warning above?

**Comment by wojtekmaj** [comment]
Please see this discussion: https://github.com/mozilla/pdf.js/issues/16061
